packages/wrangler/src/deployment-bundle/esbuild-plugins/hybrid-nodejs-compat.ts

```diff
@@ -12,22 +12,39 @@ export const nodejsHybridPlugin: () => Plugin = () => {
 	return {
 		name: "unenv-cloudflare",
 		setup(build) {
+			errorOnServiceWorkerFormat(build);
 			handleRequireCallsToNodeJSBuiltins(build);
 			handleAliasedNodeJSPackages(build, alias, external);
 			handleNodeJSGlobals(build, inject);
 		},
 	};
 };
 
+const NODEJS_MODULES_RE = new RegExp(`^(node:)?(${builtinModules.join("|")})$`);
+
+/**
+ * If we are bundling a "Service Worker" formatted Worker, imports of external modules,
+ * which won't be inlined/bundled by esbuild, are invalid.
+ *
+ * This `onResolve()` handler will error if it identifies node.js external imports.
+ */
+function errorOnServiceWorkerFormat(build: PluginBuild) {
+	build.onResolve({ filter: NODEJS_MODULES_RE }, (args) => {
+		if (build.initialOptions.format === "iife") {
+			throw new Error(
+				`Unexpected import "${args.path}" which is not valid in a Service Worker format Worker. Are you missing a default export from your Worker?`
```

Comment:
```suggestion
				`Unexpected import "${args.path}" which is not valid in a Service Worker format Worker. Are you missing a `export default { ... }` from your Worker?`
```

How about this instead?

Reply:
Yeah I like that. I was expecting a bit of back and forth on the best messaging here.
I was wondering whether we should also say something along the line of "Did you intend to create a Server Worker format Worker?" and also whether we just warn rather than error (since for some of these scenarios the Worker could actually operate fine as long as it didn't hit the line that tries to do the dynamic require).

Reply:
So if they're importing something like node: / cloudflare:, then we're pretty sure they meant a module worker (and most new workers etc are module workers today), so I'd say we lean the messaging towards module workers.

Reply:
So also instead of "which is not valid in a Service Worker format Worker" maybe we can say "which would only work in a Modules format Worker"

Reply:
I updated it to make it more ES Module oriented. PTAL